### ISSUE TITLE
PostFinance does not accept QR invoice with QR-IBAN displayed

### ISSUE THREAD

**Issue by entreprenr** [manuelbl/SwissQRBill#27]
PostFinance told one of our customers that the QR-IBAN should not be displayed to the client, instead the regular IBAN should be displayed. They rejected the QR invoice.

So this would create a QR invoice which has a Swiss QR reference, but displays a regular IBAN. Not sure if they mean that behind the QR code it uses the QR-IBAN but displays the regular IBAN on the invoice (we will have to check with PostFinance). But anyway, this is not possible with the SwissQRBill library, since it rejects the QR reference with a regular IBAN. And you cannot set an IBAN for display only, afaik.

It may very well be that PostFinance is wrong on this. In my opinion all the information that is displayed on the invoice must be exactly the same as the information behind the QR code. So we should display the QR-IBAN.

Any other experiences?

**Comment by manuelbl** [comment]
Displaying information that deviates from the information in the QR code can't be right unless I have fundamentally misunderstood the QR bill.

What is the context here: Have you issued the QR invoice and the customer was unable to make a payment to your account? If so, is the account with PF or with a bank? Or are you providing a software/service to customer and the customer has issued the QR invoice? If so, is the target a PF or a bank account?

What was PFs role when the complained about the invoice? If it was for processing a payment (e.g. via their e-banking), they wouldn't see nor care about what's printed on the invoice as they only process the QR code information.

**Comment by entreprenr** [comment]
We are providing a software to the customer and the customer sent a QR invoice to their customer who tried to pay with the QR invoice and it was rejected (but we don't know how, probably went to the Post office?). Our customer then had a telephone conversation with PostFinance where they explained that the QR-IBAN is not to be displayed on the invoice.

Of course, I agree with you that the information displayed should not deviate from the QR code.

Maybe an employee from PostFinance said it wrong or our customer misunderstood...

**Comment by entreprenr** [comment]
We have tested said QR invoice now with PostFinance and it worked, both scanning the QR code as well as entering the QR-IBAN manually. After discussing with the customer it turned out that his customer tried to enter the QR-IBAN manually with their Bank in Liechtenstein and got an error message, saying that the QR-IBAN cannot be entered manually. Apparently this Bank does not support entering QR-IBANs manually, but that is their own issue. The phone conversation of our customer with PostFinance must be some kind of misunderstanding. Closing this issue...
